projects/valtimo/zgw/src/lib/modules/documenten-api/components/documenten-api-tag-modal/documenten-api-tag-modal.component.ts

```diff
@@ -0,0 +1,153 @@
+import {TranslateModule, TranslateService} from '@ngx-translate/core';
+import {CommonModule} from '@angular/common';
+import {DocumentenApiTagService} from '../../services/documenten-api-tag.service';
+import {DocumentenApiTag} from '../../models/documenten-api-tag.model';
+
+@Component({
+  selector: 'valtimo-documenten-api-tag-modal',
+  templateUrl: './documenten-api-tag-modal.component.html',
+  styleUrls: ['./documenten-api-tag-modal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+  imports: [
+    CommonModule,
+    ModalModule,
+    TagModule,
+    TranslateModule,
+    InputModule,
+    ReactiveFormsModule,
+    ButtonModule,
+    DropdownModule,
+    ToggleModule,
+  ],
+})
+export class DocumentenApiTagModalComponent implements OnDestroy {
+  @Input() public documentDefinitionName!: string;
+  @Input() public open = false;
+
+  @Output() public closeModalEvent = new EventEmitter<DocumentenApiColumnModalTypeCloseEvent>();
+
+  public readonly tagFormGroup = this.fb.group({
+    value: this.fb.control('', [Validators.required, Validators.maxLength(50)]),
+  });
+
+  public readonly disabled$ = new BehaviorSubject<boolean>(false);
+
+  public get value(): AbstractControl<string, string> {
+    return this.tagFormGroup?.get('value');
+  }
+
+  public get invalid(): boolean {
+    return !!this.tagFormGroup?.invalid;
+  }
+
+  public get pristine(): boolean {
+    return !!this.tagFormGroup?.pristine;
+  }
+
+  private readonly _subscriptions = new Subscription();
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private readonly documentenApiTagService: DocumentenApiTagService
+  ) {}
+
+  public ngOnDestroy(): void {
+    this._subscriptions.unsubscribe();
+  }
+
+  public onClose(): void {
+    this.close();
+  }
+
+  public addTag(): void {
+    this.disable();
+
+    this.documentenApiTagService
+      .createTag(this.documentDefinitionName, this.getFormValue().value)
+      .subscribe({
+        next: () => {
+          this.enable();
+          this.closeAndRefresh();
+          this.resetForm();
+        },
+        error: () => {
+          this.enable(false);
+        },
+      });
+  }
+
+  private resetForm(): void {
+    this.tagFormGroup.patchValue({
+      value: '',
+    });
+    this.tagFormGroup.markAsPristine();
+  }
+
+  private disable(): void {
+    this.disabled$.next(true);
+    this.tagFormGroup.disable();
+  }
+
+  private enable(delay = true): void {
+    setTimeout(
+      () => {
+        this.disabled$.next(false);
+        this.tagFormGroup.enable();
+      },
+      delay ? CARBON_CONSTANTS.modalAnimationMs : 0
+    );
+  }
+
+  private close(): void {
+    this.closeModalEvent.emit('close');
+    this.resetForm();
+  }
+
+  private closeAndRefresh(): void {
+    this.closeModalEvent.emit('closeAndRefresh');
+    this.resetForm();
+  }
+
+  private getFormValue(): DocumentenApiTag {
+    return {
+      value: this.value.value,
+    };
+  }
```

Comment:
Since you're only using this in one place, why not put this value directly in the value getter?

Reply:
Removed it completely